frontend/src/concepts/k8s/pods/utils.ts

```diff
@@ -3,43 +3,54 @@ import { PodContainer } from '~/types';
 import { getPodContainerLogText } from '~/api';
 import { downloadString } from '~/utilities/string';
 
+const currentTimeStamp = new Date().toISOString();
+
 export const getPodContainers = (
   pod: PodKind | null,
 ): { containers: PodContainer[]; initContainers: PodContainer[] } => ({
   containers: pod?.spec.containers ?? [],
   initContainers: pod?.spec.initContainers ?? [],
 });
 
-export const downloadFullPodLog = async (
+export const downloadCurrentStepLog = async (
   namespace: string,
   podName: string,
   containerName: string,
+  podCompleted: boolean | undefined,
 ) =>
   getPodContainerLogText(namespace, podName, containerName).then((content) =>
-    downloadString(`${podName}-${containerName}.log`, content),
+    downloadString(
+      `${podName}-${containerName}-${podCompleted ? 'full' : currentTimeStamp}.log`,
+      content,
+    ),
   );
 
 export const downloadAllStepLogs = async (
   podContainers: PodContainer[],
   namespace: string,
-  podName: string,
+  pod: PodKind,
 ) => {
   const logPromises = podContainers
     .filter((podContainer) => podContainer !== null)
     .map(async (podContainer) => {
       const logsIndividualStep = await getPodContainerLogText(
         namespace,
-        podName,
+        pod.metadata?.name ?? '',
         podContainer.name,
       );
       return `=============
-${podContainer.name}
+${podContainer.name}  
 =============
 ${logsIndividualStep}`;
     });
-
+  const completed = (pod.status?.containerStatuses || []).every(
+    (containerStatus) => containerStatus?.state?.terminated,
+  );
   const allStepLogs = await Promise.all(logPromises);
 
   const combinedLogs = allStepLogs.join('\n');
-  downloadString(`${podName}.log`, combinedLogs);
+  downloadString(
+    `${pod.metadata?.name}-${completed ? 'full' : currentTimeStamp}.log`,
```

Comment:
Same here, after you change the `PodKind`, you don't need the optional chain here.